### PR TITLE
Ensure all header files shipped have the vk_ prefix

### PR DESCRIPTION
**Pull request by juan-lunarg**
CI will fail now if we ship a .h / .hpp file that doesn't use the correct prefix.

I also documented the precedent for this prefix and the existing integration testing.

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -15,7 +15,7 @@ static_library("vulkan_layer_settings") {
     "include/vulkan/layer/vk_layer_settings.h",
     "include/vulkan/layer/vk_layer_settings.hpp",
     "include/vulkan/layer/vk_layer_settings_ext.h",
-    "include/vulkan/utility/vul_dispatch_table.h",
+    "include/vulkan/utility/vk_dispatch_table.h",
```
Comment (juan-lunarg) [inline BUILD.gn:18]: This was the only header file not using the vk_ prefix.

File: tests/add_subdirectory/CMakeLists.txt [hunk 2]
```diff
@@ -7,28 +7,55 @@ cmake_minimum_required(VERSION 3.17)
 
 project(TEST_ADD_SUBDIRECTORY LANGUAGES C)
 
-add_library(add_subdirectory_example STATIC)
+if (NOT DEFINED VULKAN_HEADER_SOURCE_DIR)
+    message(FATAL_ERROR "VULKAN_HEADER_SOURCE_DIR not defined!")
+endif()
```
Comment (juan-lunarg) [inline tests/add_subdirectory/CMakeLists.txt:12]: Error out quickly with a clear error message

File: tests/add_subdirectory/CMakeLists.txt [hunk 3]
```diff
@@ -7,28 +7,55 @@ cmake_minimum_required(VERSION 3.17)
...
+foreach(header IN LISTS public_headers)
+    get_filename_component(header ${header} "NAME")
+    message(DEBUG "Checking ${header}")
+    if (${header} MATCHES "^vk_")
+        message(DEBUG "Correct naming convention!")
+    else()
+        # EX: "Invalid file name! vk_dispatch_table.h"
+        message(FATAL_ERROR "Invalid file name! ${header}")
+    endif()
+endforeach()
```
Comment (juan-lunarg) [inline tests/add_subdirectory/CMakeLists.txt:36]: This is how I ensure the files we ship have the correct prefix.

File: tests/add_subdirectory/vk_dispatch_table.c [hunk 4]
```diff
@@ -3,21 +3,23 @@
...
+// NOLINTBEGIN
+
+PFN_vkVoidFunction VKAPI_PTR local_vkGetInstanceProcAddr(VkInstance instance, const char *pName) {
     (void)instance;
     (void)pName;
     return NULL;
 }
 
-PFN_vkVoidFunction local_vkGetDeviceProcAddr(VkDevice device, const char *pName) {
+PFN_vkVoidFunction VKAPI_PTR local_vkGetDeviceProcAddr(VkDevice device, const char *pName) {
```
Comment (juan-lunarg) [inline tests/add_subdirectory/vk_dispatch_table.c:16]: FYI @charles-lunarg I had to add `VKAPI_PTR` otherwise my build failed to compile for x86

Comment (charles-lunarg) [reply tests/add_subdirectory/vk_dispatch_table.c:14]: That sounds right, any function pointer getting passed around on windows need VKAPI_PTR to get the correct calling convention.

**Review by charles-lunarg — APPROVED** [review]

**Pull request merged**